apps/client/src/common/hooks/useEventAction.ts

```diff
@@ -152,6 +153,54 @@ export const useEventAction = () => {
     [_updateEventMutation],
   );
 
+  /**
+   * Calls mutation to edit multiple events
+   * @private
+   */
+  const _batchUpdateEventsMutation = useMutation({
+    mutationFn: requestBatchPutEvents,
+    onMutate: async ({ ids, data }) => {
+      // cancel ongoing queries
+      await queryClient.cancelQueries([RUNDOWN_TABLE_KEY]);
+
+      // Snapshot the previous value
+      const previousEvents = queryClient.getQueryData([RUNDOWN_TABLE_KEY]) as OntimeRundown;
+
+      const updatedEvents = previousEvents.map((event) => {
+        const isEventEdited = ids.includes(event.id);
+
+        if (isEventEdited && isOntimeEvent(event)) {
+          return {
+            ...event,
+            // spread new changed properties, overwriting old event properties
+            ...data,
+          };
+        }
+
+        return event;
+      });
+
+      queryClient.setQueryData(RUNDOWN_TABLE, updatedEvents);
+
+      // Return a context with the previous and new events
+      return { previousEvent: previousEvents };
```

Comment:
mega nitpick: i tend to think it is nicer to use the shorthand `return { previousEvent };`
what do you think?

Reply:
Left over from renaming. Done